machine_learning/similarity_search.py

```diff
@@ -0,0 +1,109 @@
+"""
+Simularity search is a search algorithm for finding the nearest vector from
+vectors, used in natural language processing.
+In this algorithm, it calculates distance with euclidean distance and
+returns a list containing two data for each vector:
+    1. the nearest vector
+    2. distance between the vector and the nearest vector
+"""
+import numpy as np
+import math
+
+
+def euclidean(input_a, input_b):
```

Comment:
Add type hints. See: https://docs.python.org/3/library/typing.html

Reply:
input_a and input_b can be integer, float or list containing integer or float. Do you have any recommendations for it?

Reply:
```python
import math
from typing import Union, TypeVar

# define a type variable that can be int or float
IntOrFloat = TypeVar('IntOrFloat', int, float)
# IntFloatOrList states that the value must be an int, float or a list of int or float
IntFloatOrList = Union[IntOrFloat, List[IntOrFloat]]

def euclidean(input_a: IntFloatOrList, input_b: IntFloatOrList):
    ...
```

Reply:
Got it :)